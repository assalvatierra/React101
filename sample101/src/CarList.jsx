import React from 'react';

class CarListClass extends React.Component{

    render(){
        const isready = true;
        let datatable = (isready ? <p>Ready</p> : <p>Loading</p>);
        return(
            <div>
            <h1>Car List class</h1>    
            {datatable}
            </div>
            
        );
    }
}

const CarList = CarListClass;
export default CarList;