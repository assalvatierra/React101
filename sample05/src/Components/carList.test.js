import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';
import CarList from './carList';

let container = null;
beforeEach( ()=> {
    container = document.createElement('div');
    document.body.appendChild(container);
});

afterEach( ()=> {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
})

it('render car list', async() => {
    const fakeList = 
        [{id:0, name:'abel', remarks:'test'},
        {id:1, name:'elvie', remarks:'test2'}];

    jest.spyOn(global, 'fetch').mockImplementation(()=> 
    Promise.resolve({
        json: ()=>Promise.resolve(fakeList)
    })   
    );

    await act( async()=>{
        render(<CarList />, container);
    });

    expect(container.textContent).toContain('abel');
    expect(container.textContent).toContain('elvie');

    expect(
        pretty(container.innerHTML)
    ).toMatchSnapshot();

    global.fetch.mockRestore();

});