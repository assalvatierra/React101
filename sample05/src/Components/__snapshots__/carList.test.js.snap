// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render car list 1`] = `
"<div>
  <table>
    <thead>
      <tr>
        <td>id</td>
        <td>Name</td>
        <td>Remarks</td>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>abel</td>
        <td>test</td>
      </tr>
      <tr>
        <td>elvie</td>
        <td>test2</td>
      </tr>
    </tbody>
  </table>
</div>"
`;
